core/org.wso2.carbon.ui/src/main/java/org/wso2/carbon/ui/CarbonUIUtil.java

```diff
@@ -502,4 +537,70 @@ public static String removeTenantSpecificStringsFromURL(String requestURL) {
     private static Object getDefaultHomePageProductParam() {
         return getProductParam(CarbonConstants.PRODUCT_XML_WSO2CARBON + CarbonConstants.DEFAULT_HOME_PAGE);
     }
+
+    /**
+     * Returns absolute URL of admin console webapp for given relative path
+     * if IS_RESOLVE_ABSOLUTE_URLS_ENABLED config is enabled.
+     *
+     * @param context       Webapp context root of the Carbon webapp.
+     * @param relativePath  Relative path of the Carbon webapp
+     * @param request       Request that used to redirect.
+     * @return absolute URL of admin console webapp for given relative path.
+     */
+    public static String resolveAdminConsoleBaseURL(String context, String relativePath, HttpServletRequest request) {
+
+        if (isResolveAbsoluteURLsEnabled()) {
+
+            // Removing any tailing "/" in the context.
+            context = getAdminConsoleBaseURL(context);
+            if (context.endsWith("/")) {
+                context = context.substring(0, context.length() - 1);
+            }
+
+            // Remove any tailing "/carbon" in context to build base URL.
+            if (context.endsWith("/carbon")) {
+                context = context.substring(0, context.length() - 7);
+            }
+
+            // Build relative path starting from root context.
+            List<String> splitPathList = new ArrayList<>(Arrays.asList(
+                    request.getContextPath().concat(request.getServletPath()).split("/")));
+            splitPathList.remove(0);
+            // If the request is a base URL, add the carbon as the root context.
+            if (splitPathList.isEmpty()) {
+                splitPathList.add("carbon");
+            }
+
+            // Replace ".." with the node of path directory.
+            int index = 0;
+            while (relativePath.contains("..")) {
+                relativePath = relativePath.replaceFirst("..", splitPathList.get(index));
+                index++;
+            }
+
+            // Add "/", if relative path is not starting with.
+            if (relativePath.charAt(0) != '/') {
+                relativePath = "/" + relativePath;
+            }
+        }
+
+        return context + relativePath;
+    }
+
+    /**
+     * Returns whether resolving absolute URL config is enabled or not.
+     *
+     * @return Resolving absolute URL config is enabled.
+     */
+    public static boolean isResolveAbsoluteURLsEnabled() {
+
+        String  isResolveAbsoluteURLsEnabled = CarbonUIServiceComponent.getServerConfiguration()
```

Comment:
```suggestion
        String isResolveAbsoluteURLsEnabled = CarbonUIServiceComponent.getServerConfiguration()
```

Reply:
fixed in https://github.com/wso2/carbon-kernel/pull/4371/commits/eb3c289f3f85e233d5e4c73d253bb9f4f6077b8b